widgetastic_manageiq/__init__.py

```diff
@@ -4051,3 +4051,166 @@ def data_for_timestamp(self, timestamp):
+
+    @property
+    def list_all_items_locators(self):
+        """Returns list of item locators for each of the items."""
+        return self.browser.elements(self.ITEMS_LOCATOR)
+
+    @property
+    def list_selected_items_locators(self):
+        """Returns list of item locators for currently selected items."""
+        return self.browser.elements(self.SELECTED_ITEMS_LOCATOR)
+
+    def _get_all_options(self):
+        """ Method to return dict of all the item webelement locators and text"""
+        return {element.text: element for element in self.list_all_items_locators}
+
+    def read(self):
+        """Method to read elements that are currently selected."""
+        return self.list_selected_items
+
+    def fill(self, values, partial_match=False):
+        """Click/Select multiple the items in the multiselectbox
+
+        Args:
+            items: 'list' of items of type 'str' to be matched
+            partial_match: if you want to use partial text matching. Default:False
+        Returns: True if all values were found and clicked. False if key_error_count != 0.
+         """
+        if values == self.list_selected_items or values == []:
+            return False
+        key_error_count = 0
+        if values:
+            for item in values:
+                try:
+                    self.select_by_text(item, partial_match=partial_match)
+                except KeyError:
+                    key_error_count += 1
+                    self.logger.debug("Could not find %s in %s", item, self.list_items)
+        return key_error_count == 0  # Meaning filling was successful
+
+    def select_by_text(self, text, partial_match=False):
+        """Select/click option using text
+
+        Args:
+        text(str): text to be matched
+        partial_match: if you want to use partial text matching. Default:False
+         """
+        try:
+            if text and not partial_match:
+                self.logger.info("Not using partial text matching to find %s", text)
+                self._get_all_options()[text].click()
+
+            elif text and partial_match:
+                self.logger.info("Using partial text matching to find %s", text)
+                all_options = self._get_all_options()
+                for key in all_options:
+                    if text in key:
+                        all_options[key].click()
+                        break
+        except KeyError:
+            raise ItemNotFound("%s was not found in %s", text, self.list_items)
+
+
+class InfraMappingTreeView(Widget):
+    """Widget to capture/interact with the InfraMappingTreeView in Infra Mapping Wizard for v2v.
+
+    Args:
+        tree_class(str): Class name of the div representing this TreeView
+    """
+    ROOT = ParametrizedLocator('.//div[contains(@class,{@tree_class|quote})]/ul/li')
+    ROOT_ITEMS = '//span[contains(@class,"expand-icon")]/..'
+    ITEMS_LOCATOR = ('/span/../text()[not(contains(.,"Default")) '
```

Comment:
Because I could not get it to work that way. :( I will try to debug it with Mike today. self.browser.elements without parent=self was not yielding correct output, and removing self.root_browser was only getting me 1 element instead of the list.

Reply:
**REQUIRED**
Please change your xpaths, it's the reason. I strongly recommend you to look implementation of `BootstrapTreeview`. Your tree is very similar. You can even try to inherit it and override some methods and attributes.

Reply:
I tried that briefly when I began to implement it, but I could not see a way to implement off of BootstrapTreeview but I will reconsider.

Reply:
At least look on xpaths of `BootstrapTreeview`, except `ROOT` they are should be the same.

Reply:
@quarckster can you point me to a use case in manageiq where I can find other tree view? so I can compare the locators in those and my treeview!

Reply:
Got it to work as 
```
self.browser.elements(self.ROOT_ITEMS)
self.browser.elements(self.ITEMS_LOCATOR)
```